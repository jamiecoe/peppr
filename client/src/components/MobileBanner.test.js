import React from 'react';
import { shallow } from 'enzyme';
import Banner from './MobileBanner';

describe('MobileBanner', () => {
  const props = { title: 'My recipes' };
  const banner = shallow(<Banner {...props} />);

  it('renders properly', () => {
    expect(banner).toMatchSnapshot();
  });

  it('displays the title from props', () => {
    expect(banner.find('.mobileBanner__title').text()).toEqual(props.title);
  });
});
