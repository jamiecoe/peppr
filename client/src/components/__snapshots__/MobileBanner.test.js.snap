// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileBanner renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Banner
    title="My recipes"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2
        className="mobileBanner__title"
>
        My recipes
</h2>,
      "className": "mobileBanner",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "My recipes",
        "className": "mobileBanner__title",
      },
      "ref": null,
      "rendered": "My recipes",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2
          className="mobileBanner__title"
>
          My recipes
</h2>,
        "className": "mobileBanner",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My recipes",
          "className": "mobileBanner__title",
        },
        "ref": null,
        "rendered": "My recipes",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
